news/php/news_article_gallery_feed.php

```diff
@@ -36,27 +36,54 @@ function create_news_article_gallery_feed($categories, $galleryStyle, $myBethel,
 //    $arrayOfNewsAndStories = get_xml($_SERVER["DOCUMENT_ROOT"] . "/_shared-content/xml/news-and-stories.xml", $categories, "inspect_news_article");
     $arrayOfNewsAndStories = sort_by_date($arrayOfNewsAndStories);
 
+    $includeStories = false;
+    $includeNews = false;
+
+    if( $newsOrStories == "Both" ) {
+        $includeStories = true;
+        $includeNews = true;
+    } elseif ( $newsOrStories == "News" ) {
+        $includeNews = true;
+    } else {
+        $includeStories = true;
+    }
+
     $threeStories = array();
-    foreach( $arrayOfNewsAndStories as $index => $newsAndStory){
+    foreach ($arrayOfNewsAndStories as $index => $newsAndStory) {
         $id = $newsAndStory['id'];
+        if( $includeNews ) {
+            // if its the Homepage Top Feature, skip any that aren't tagged as homepage
+            if( $galleryStyle == 'Homepage Top Feature' && !$newsAndStory['homepage-article'] )
+                continue;
+
+            if( $newsAndStory['article-type'] == 'News' and !in_array($id, $GLOBALS['stories-already-used'])) {
+                $newsAndStory['gallery-image'] = srcset($newsAndStory['image-path'], false, true, '', $newsAndStory['title']);
 
-        // if its the Homepage Top Feature, skip any that aren't tagged as homepage
-        if( $galleryStyle == 'Homepage Top Feature' && !$newsAndStory['homepage-article'] )
-            continue;
+                // don't use this story on this page again
+                array_push($GLOBALS['stories-already-used'], $id);
 
-        if( $newsAndStory['article-type'] == 'Story' and !in_array($id, $GLOBALS['stories-already-used'])){
-            $newsAndStory['gallery-image'] = srcset($newsAndStory['image-path'], false, true, '', $newsAndStory['title']);
+                array_push($threeStories, $newsAndStory);
+                unset($arrayOfNewsAndStories[$index]);
+            }
+        }
+        if( $includeStories ) {
+            // if its the Homepage Top Feature, skip any that aren't tagged as homepage
+            if( $galleryStyle == 'Homepage Top Feature' && !$newsAndStory['homepage-article'] )
+                continue;
 
-            // don't use this story on this page again
-            array_push($GLOBALS['stories-already-used'], $id);
+            if( $newsAndStory['article-type'] == 'Story' and !in_array($id, $GLOBALS['stories-already-used'])){
+                $newsAndStory['gallery-image'] = srcset($newsAndStory['image-path'], false, true, '', $newsAndStory['title']);
```

Comment:
line 75 should have a comment. I assume this is specific to the gallery news feed, but its worth mentioning why we do extra image generation (since a lot of it is done in the inspect_news method

Reply:
I'm not sure why we do this-- it was already there, so I just kept it in when I made my edits

Reply:
yeah its probably worth fixing since you are touching it :) Thanks!